components/src/dynamo/sglang/request_handlers/embedding_handler.py

```
@@ -0,0 +1,80 @@
+        logging.debug(f"Embedding request: {request}")
+
+        # Parse the embedding request - should only receive EmbeddingRequest format
+        embedding_request = EmbeddingRequest(**request)
+
+        # Handle different input types
+        if isinstance(embedding_request.input, str):
+            prompt = embedding_request.input
+        elif isinstance(embedding_request.input, list):
+            prompt = embedding_request.input
+        else:
+            raise ValueError(f"Invalid input type: {type(embedding_request.input)}")
```

Ruff (0.13.3)

Prefer `TypeError` exception for invalid type

(TRY004)